### PR TITLE
Add support of ssh authentications defined in compose file or via cli…

### LINKED ISSUE #7025: Enable ssh option as build parameter for buildkit
**Is your feature request related to a problem? Please describe.**
When building a container we need access to other Resources like ssh and dont want to expose a private key in the container. 

**Describe the solution you'd like**
Support of `--ssh` in the build process that already works in buildkit.

**Additional context**
I tried `1.25.0-rc4`, but it does not support `--ssh` option. Tried the same as in bug 6440:

```
export COMPOSE_DOCKER_CLI_BUILD=1
export DOCKER_BUILDKIT=1
docker-compose build --ssh default my_image
```
This would be a major security improvement and save us so much pain.

Thanks,
bert

### PR DESCRIPTION
**Pull request by glours**
… flags

**What I did**

- Add support of an `ssh` flag for the `build` command
- Generate an SSH Agent Provider from both cli flag and Compose file configuration

**Related issue**
fix #7025

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**
image

### CODE REVIEW

**Review by ndeloof — APPROVED** [review]

**Review by ndeloof — APPROVED** [review]

**Comment by ndeloof** [comment]
🎉

**Comment by levrik** [comment]
Is there any expected timeline when this will hit Docker on Linux and macOS?

**Comment by aleksandrs-ledovskis** [comment]
It's possible to already use it on macOS and Linux if compiled from repo's source.

For those using Homebrew, the command is `$ brew install --head docker-compose`.

File: docs/reference/compose_build.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Build or rebuild services
 | `--progress` | `string` | `auto` | Set type of progress output (auto, tty, plain, quiet) |
 | `--pull` |  |  | Always attempt to pull a newer version of the image. |
 | `-q`, `--quiet` |  |  | Don't print anything to STDOUT |
+| `--ssh` | `string` |  | Set SSH authentications used when building service images. (use 'default' for using you default SSH Agent) |
```
Comment (aleksandrs-ledovskis) [inline docs/reference/compose_build.md:15]: `s/using you/using your/`

Comment (aleksandrs-ledovskis) [reply docs/reference/compose_build.md:15]: Hivemind with https://github.com/docker/compose/pull/9334 🙂

**Comment by nleroy917** [comment]
Hello,

I compiled from source (`brew install --head docker-compose`), but the `--ssh` flag still seems unknown:

```sh
docker-compose up --build --ssh default 
unknown flag: --ssh
```

Version seems up to date:

```
docker-compose -v                      
Docker Compose version HEAD-7969667
```

Any help would be appreciated!

**Comment by glours** [comment]
@nleroy917 `--ssh` flag can be use with `docker compose build` but is not available with `docker compose up` command

**Comment by nleroy917** [comment]
🤦🏻‍♂️ That makes sense. Thank you!

**Comment by IT-Ideas** [comment]
Hello all!
If I am not mistaken this only take one pair key/value as argument of --ssh, right?
We are in a situation where we would have like to pass two different ssh keys.
Would there be any plans to manage this use case?

Thanks!

**Comment by ndeloof** [comment]
@IT-Ideas ssh configuration should be set by your `compose.yaml` file inside the `build` section (https://docs.docker.com/compose/compose-file/build/#ssh). `--ssh` flag is just a helper for simplest use-cases, imho we should not expose such flags which are redundant with the yaml model

**Comment by IT-Ideas** [comment]
Hi @ndeloof,
I agree with you and I actually tested it, but the path to the key was then not working although it worked when passing it through the command.
The path beeing ~/.ssh/keyname.

Thanks 🙂

**Comment by IT-Ideas** [comment]
@ndeloof  Maybe I did a mistake, I'll have one more try!

**Comment by ndeloof** [comment]
please open a dedicated issue if you can reproduce this error and attach a sample compose.yaml file, thanks

**Pull request merged**